internal/svc/cerbos_svc.go

```diff
@@ -61,6 +61,9 @@ func (cs *CerbosService) ResourcesQueryPlan(ctx context.Context, request *reques
 	response, err := cs.eng.ResourcesQueryPlan(logging.ToContext(ctx, log), input)
 	if err != nil {
 		log.Error("Resources query plan request failed", zap.Error(err))
+		if errors.Is(err, engine.ErrNoSuchKey) {
+			return nil, status.Errorf(codes.Internal, err.Error())
```

Comment:
In this case the request is bad, so we should return a 4xx status

```suggestion
			return nil, status.Errorf(codes.InvalidArgument, "Bad request: %v", err)
```